Update and fix failing standalone API Integration Tests

|Component|
|---|
 | API Integration tests |

This issue is related to the epic issue #10137. In that epic, new manager standalone API integration tests are told to be implemented.

With the changes requested in the related epic issue, specifically in #10592, new marks are going to be added to all API integration tests (but RBAC). After this addition, tests should be passed using both marks to use both docker environment.

In this issue, we should update and fix all API Integration Tests with unexpected fails in the standalone mode. No tests should be failing in the cluster environment since it was our default environment before this epic issue.

## Checks

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)
<!-- If changes are made to any of the following components, uncomment the corresponding line 
- [ ] **Integration tests** without failures for API configuration (`/wazuh-qa/tests/integration/test_api/test_config/`)
- [ ] **System tests** for agent enrollment process (`/wazuh-qa/tests/system/test_cluster/test_agent_enrollment`)
- [ ] **System tests** for agent info sync process in cluster (`/wazuh-qa/tests/system/test_cluster/test_agent_info_sync`)
- [ ] **System tests** for agent key polling (`/wazuh-qa/tests/system/test_cluster/test_agent_key_polling`)
- [ ] **System tests** for JWT invalidation (`/wazuh-qa/tests/system/test_jwt_invalidation`)
-->

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

## Issue update

### Agent GET

Error:

```
FAILED test_agent_GET_endpoints.tavern.yaml::GET /agents/{agent_id}/config/{component}/{configuration}
```

```
Differing items:
  {'error': 1116} != {'error': <tavern.util.loader.IntSentinel object at 0x7f69f3e91550>}
  Left contains 4 more items:
  {'dapi_errors': {'node01': {'error': 'Requested component configuration does '
                                       'not exist: com:cluster'}},
   'detail': 'Requested component configuration does not exist: com:cluster',
   'remediation': 'Please, visit the official documentation '...
  
  ...Full output truncated (15 lines hidden), use '-vv' to show
```

This test should not be done in a standalone environment as it is normal that the cluster configuration is not returned.

I have added a new test case in order to test the `GET /agents/{agent_id}/config/{component}/{configuration}` endpoint only in a cluster environment.

This error is going to be solved in this issue as it is considered a test-related problem.

--- a/api/test/integration/env/configurations/experimental/manager/healthcheck/healthcheck.py
+++ b/api/test/integration/env/configurations/experimental/manager/healthcheck/healthcheck.py
@@ -8,4 +8,4 @@ from healthcheck_utils import get_manager_health_base
 if __name__ == "__main__":
     # Workers are not needed in this test, so the exit code is set to 0 (healthy).
     exit(os.system("grep -q 'wazuh-modulesd:syscollector: INFO: Evaluation finished.' /var/ossec/logs/ossec.log")
-         or get_manager_health_base()) if socket.gethostname() == 'wazuh-master' else exit(0)
+         or get_manager_health_base(env_mode=sys.argv[1] if len(sys.argv) > 1 else None)) if socket.gethostname() == 'wazuh-master' else exit(0)

--- a/api/test/integration/env/configurations/syscheck/manager/healthcheck/healthcheck.py
+++ b/api/test/integration/env/configurations/syscheck/manager/healthcheck/healthcheck.py
@@ -9,4 +9,4 @@ if __name__ == "__main__":
     # Workers are not needed in this test, so the exit code is set to 0 (healthy).
     exit(os.system(
         "grep -q 'wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.' /var/ossec/logs/ossec.log")
-         or get_manager_health_base()) if socket.gethostname() == 'wazuh-master' else exit(0)
+         or get_manager_health_base(env_mode=sys.argv[1] if len(sys.argv) > 1 else None)) if socket.gethostname() == 'wazuh-master' else exit(0)

--- a/api/test/integration/env/configurations/syscollector/manager/healthcheck/healthcheck.py
+++ b/api/test/integration/env/configurations/syscollector/manager/healthcheck/healthcheck.py
@@ -9,4 +9,4 @@ if __name__ == "__main__":
     # Workers are not needed in this test, so the exit code is set to 0 (healthy).
     exit(os.system(
         "grep -q 'wazuh-modulesd:syscollector: INFO: Evaluation finished.' /var/ossec/logs/ossec.log")
-         or get_manager_health_base()) if socket.gethostname() == 'wazuh-master' else exit(0)
+         or get_manager_health_base(env_mode=sys.argv[1] if len(sys.argv) > 1 else None)) if socket.gethostname() == 'wazuh-master' else exit(0)

--- a/api/test/integration/env/configurations/vulnerability/manager/healthcheck/healthcheck.py
+++ b/api/test/integration/env/configurations/vulnerability/manager/healthcheck/healthcheck.py
@@ -65,7 +65,7 @@ if __name__ == '__main__':
         agents_node = get_healthcheck_state()
 
     # Create a checks list with the manager health base at first
-    checks_list = [get_manager_health_base()]
+    checks_list = [get_manager_health_base(env_mode=sys.argv[1] if len(sys.argv) > 1 else None)]
 
     expected_log = "grep -q 'wazuh-modulesd:vulnerability-detector: INFO: (5471): Finished vulnerability assessment " \
                    "for agent '\\''{}'\\''' /var/ossec/logs/ossec.log"

--- a/api/test/integration/env/docker-compose.yml
+++ b/api/test/integration/env/docker-compose.yml
@@ -92,6 +92,7 @@ services:
       - nginx-lb
       - wazuh-agent2
     depends_on:
+      - wazuh-agent1
       - nginx-lb
 
   wazuh-agent3:
@@ -111,6 +112,7 @@ services:
       - nginx-lb
       - wazuh-agent3
     depends_on:
+      - wazuh-agent2
       - nginx-lb
 
   wazuh-agent4:
@@ -130,6 +132,7 @@ services:
       - nginx-lb
       - wazuh-agent4
     depends_on:
+      - wazuh-agent3
       - nginx-lb
 
   wazuh-agent5:
@@ -150,6 +153,7 @@ services:
       - wazuh-agent5
       - agent_old
     depends_on:
+      - wazuh-agent4
       - nginx-lb
 
   wazuh-agent6:
@@ -170,6 +174,7 @@ services:
       - wazuh-agent6
       - agent_old
     depends_on:
+      - wazuh-agent5
       - nginx-lb
 
   wazuh-agent7:
@@ -190,6 +195,7 @@ services:
       - wazuh-agent7
       - agent_old
     depends_on:
+      - wazuh-agent6
       - nginx-lb
 
   wazuh-agent8:
@@ -210,6 +216,7 @@ services:
       - wazuh-agent8
       - agent_old
     depends_on:
+      - wazuh-agent7
       - nginx-lb
 
   nginx-lb:

--- a/api/test/integration/test_agent_GET_endpoints.tavern.yaml
+++ b/api/test/integration/test_agent_GET_endpoints.tavern.yaml
@@ -1489,28 +1489,6 @@ stages:
               request_timeout: !anyint
               max_restart_lock: !anyint
 
-  - name: Request-Com-Cluster (Manager)
-    request:
-      verify: False
-      url: "{protocol:s}://{host:s}:{port:d}/agents/000/config/com/cluster"
-      method: GET
-      headers:
-        Authorization: "Bearer {test_login_token}"
-    response:
-      status_code: 200
-      json:
-        error: !anyint
-        data:
-          bind_addr: !anystr
-          disabled: !anybool
-          hidden: !anystr
-          key: !anystr
-          name: !anystr
-          node_name: !anystr
-          node_type: !anystr
-          nodes: !anything
-          port: !anyint
-
   - name: Request-Logcollector-Localfile (Manager)
     request:
       verify: False
@@ -1929,6 +1907,36 @@ stages:
       status_code: 400
       json:
         <<: *error_spec
+
+---
+test_name: GET /agents/{agent_id}/config/{component}/{configuration} (only cluster configuration)
+
+marks:
+  - cluster
+
+stages:
+
+- name: Request-Com-Cluster (Manager)
+  request:
+    verify: False
+    url: "{protocol:s}://{host:s}:{port:d}/agents/000/config/com/cluster"
+    method: GET
+    headers:
+      Authorization: "Bearer {test_login_token}"
+  response:
+    status_code: 200
+    json:
+      error: !anyint
+      data:
+        bind_addr: !anystr
+        disabled: !anybool
+        hidden: !anystr
+        key: !anystr
+        name: !anystr
+        node_name: !anystr
+        node_type: !anystr
+        nodes: !anything
+        port: !anyint
 
 ---
 test_name: GET /agents/{agent_id}/group/is_sync

